src/modules/Builder/components/Popups/SaveAndPublishProcessPopup/SaveAndPublishProcessPopup.tsx

```diff
@@ -33,7 +40,52 @@ export const SaveAndPublishProcessPopup = ({
     onClose();
   };
 
-  if (!step) return null;
+  useEffect(() => {
```

Comment:
Yes, that's right! I am on a way for helping :)

It's good idea to reduce a phantom logic in from `useEffect` and move it into the proper callback, feel free to ask details :).
The right place is in the hook `useSaveAndPublishSetup` from `SaveAndPublish.hooks.ts` (I guess it'll be renamed to `SaveAndPublish.hooks.tsx` to support the html tags)

```
 ...
  // The folowing constants should be places on top of the hook:
  const getAppletData = useAppletDataFromForm();
  const name = getAppletData()?.displayName;
  // ...

  // Here the function with all the logic used in this `useEffect`, I a bit simplified the nesting via early return
  const checkIfToShowSuccessBanner = () => {
    // If there is any visible banner warning the user they haven't made changes,
    // remove it before showing the success banner.
    dispatch(
      banners.actions.removeBanner({
        key: 'AppletWithoutChangesBanner',
      }),
    );

    dispatch(
      banners.actions.addBanner({
    key: 'SaveSuccessBanner',
    bannerProps: {
    children: (
    <Trans i18nKey="appletSavedAndPublished">
      <>
        Your applet
        <strong>
          <>{{ name }}</>
        </strong>
        was successfully uploaded!
      </>
    </Trans>
    ),
    'data-testid': 'dashboard-applets-save-success-banner',
    },
      }),
    );

    handlePublishProcessOnClose();
  };
```

And you also need to use this function in places: row 451 and 465.

Moreover, no need to set `SaveAndPublishSteps.Success` anymore in row 309 `responseStatus === 'success' && setPublishProcessStep(SaveAndPublishSteps.Success);`, please get rid of it.
Thus, no need the code:
```
if (
    !step ||
    // If the user has successfully published, we trigger a banner using the
    // dispatch above instead of rendering a modal.
    step === SaveAndPublishSteps.Success
  )
    return null;
```

Reply:
Thank you so much for the direction! Updated